galaxy_ng/app/api/ui/views/auth.py

```diff
@@ -19,10 +22,24 @@
 )
 
 
+class _CSRFSessionAuthentication(SessionAuthentication):
+    """CSRF-enforcing version of a SessionAuthentication class."""
+
+    def authenticate(self, request):
+        user = getattr(request._request, "user", None)
+        self.enforce_csrf(request)
+        return user, None
+
+
 class LoginView(api_base.GenericAPIView):
     serializer_class = LoginSerializer
+    authentication_classes = (_CSRFSessionAuthentication,)
     permission_classes = (RestrictOnCloudDeployments,)
 
+    @method_decorator(ensure_csrf_cookie)
+    def get(self, request, *args, **kwargs):
```

Comment:
To verify I understand correctly, the _CSRFSessionAuthentication Authentication class is used to enfore existence of a csrf token, even for unauthenticated or "safe" operations.

The 'get' method here requires user, auth, and csrf token and returns an empty response body. But the ensure_csrf_cookie() decorator will set the right response headers to include any needed fields for csrf
(either in the django session, or a csrf token in a cookie).

Reply:
See my comment below. For unauthenticated and unsafe. Specifically for `POST` method to `.../auth/login`.